processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -525,15 +362,15 @@ public int size()
       return size;
     }
 
-    protected T _get(ByteBuffer copyValueBuffer, int startOffset, int endOffset)
+    T bufferedIndexedGet(ByteBuffer copyValueBuffer, int startOffset, int endOffset)
     {
       final int size = endOffset - startOffset;
-      if (startOffset == endOffset) {
+      lastReadSize = size;
+      if (size == 0) {
         return null;
       }
+      copyValueBuffer.clear();
```

Comment:
Don't understand this, to what previous state it need to return? I added this `clear()` because relaxed `ObjectStrategy.fromByteBuffer()` contract, it is allowed to reset the limit now. So if the limit is changed, `position()` in the line 373 may throw exception, if position is set beyond the limit. `clear()` sets the limit to the maximum possible, the capacity.